pkg/kclient/ingress_test.go

```diff
@@ -0,0 +1,236 @@
+package kclient
+
+import (
+	"testing"
+
+	"github.com/pkg/errors"
+	extensionsv1 "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	ktesting "k8s.io/client-go/testing"
+)
+
+func TestCreateIngress(t *testing.T) {
+
+	tests := []struct {
+		name        string
+		ingressName string
+		wantErr     bool
+	}{
+		{
+			name:        "Case: Valid ingress name",
+			ingressName: "testIngress",
+			wantErr:     false,
+		},
+		{
+			name:        "Case: Invalid ingress name",
+			ingressName: "",
+			wantErr:     true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// initialising the fakeclient
+			fkclient, fkclientset := FakeNew()
+			fkclient.Namespace = "default"
+
+			objectMeta := CreateObjectMeta(tt.ingressName, "default", nil, nil)
+
+			fkclientset.Kubernetes.PrependReactor("create", "ingresses", func(action ktesting.Action) (bool, runtime.Object, error) {
+				if tt.ingressName == "" {
+					return true, nil, errors.Errorf("ingress name is empty")
+				}
+				ingress := extensionsv1.Ingress{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: tt.ingressName,
+					},
+				}
+				return true, &ingress, nil
+			})
+
+			IngressSpec := GenerateIngressSpec(IngressParameter{ServiceName: tt.ingressName})
+			createdIngress, err := fkclient.CreateIngress(objectMeta, *IngressSpec)
+
+			// Checks for unexpected error cases
+			if !tt.wantErr == (err != nil) {
+				t.Errorf("fkclient.CreateIngress unexpected error %v, wantErr %v", err, tt.wantErr)
+			}
+
+			if err == nil {
+				if len(fkclientset.Kubernetes.Actions()) != 1 {
+					t.Errorf("expected 1 action, got: %v", fkclientset.Kubernetes.Actions())
+				} else {
+					if createdIngress.Name != tt.ingressName {
+						t.Errorf("ingress name does not match the expected name, expected: %s, got %s", tt.ingressName, createdIngress.Name)
+					}
+				}
+			}
+
+		})
+	}
+}
+
+func TestListIngresses(t *testing.T) {
+	tests := []struct {
```

Comment:
since the function takes label selector as an input, the struct should at least contain the below fields

```
type args struct {
		labelSelector string
}
tests := []struct {
                args    args
		wantIngress   []extensionsv1.Ingress
		wantErr bool
}
```
the ingressName should be not be used and should instead be a part of ingress as the length of the returned list can be greater than 1

Reply:
updated the testcase.